homeassistant/components/kmtronic/switch.py

```diff
@@ -0,0 +1,88 @@
+"""KMtronic Switch integration."""
+from datetime import timedelta
+import logging
+
+import aiohttp
+import async_timeout
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DATA_HOST, DATA_HUB, DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Config entry example."""
+    hub = hass.data[DOMAIN][entry.entry_id][DATA_HUB]
+    host = hass.data[DOMAIN][entry.entry_id][DATA_HOST]
+    await hub.async_get_relays()
+
+    async def async_update_data():
+        try:
+            async with async_timeout.timeout(10):
+                await hub.async_update_relays()
+        except aiohttp.client_exceptions.ClientConnectorError as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=f"{MANUFACTURER} {hub.name}",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=30),
+    )
+
+    async_add_entities(
+        [KMtronicSwitch(coordinator, host, relay) for relay in hub.relays]
+    )
+
+
+class KMtronicSwitch(CoordinatorEntity, SwitchEntity):
+    """KMtronic Switch Entity."""
+
+    def __init__(self, coordinator, host, relay):
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self._host = host
+        self._relay = relay
+
+    @property
+    def available(self) -> bool:
+        """Return whether the entity is available."""
+        return self.coordinator.last_update_success
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return f"Relay{self._relay.id}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID of the entity."""
+        return f"{self._host}_relay{self._relay.id}"
```

Comment:
Is `self.host` stable?

https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
If not use the config entry id

Reply:
it must be stable, otherwise there is no way of reaching the device :( there is no discovery mechanism...